### ISSUE TITLE
EXPORTing symbols is inconsistent

### ISSUE THREAD

**Issue by ISSOtm** [rednex/rgbds#421]
```
issotm@sheik-kitty tmp.661JcLpBHt% cat a.asm
SECTION "a", ROM0[0]
    db ANSWER
issotm@sheik-kitty tmp.661JcLpBHt% cat b.asm
ANSWER = 42
EXPORT ANSWER

SECTION "b", ROM0
issotm@sheik-kitty tmp.661JcLpBHt% rgbasm -o a.o a.asm
issotm@sheik-kitty tmp.661JcLpBHt% rgbasm -o b.o b.asm
issotm@sheik-kitty tmp.661JcLpBHt% rgblink -o out.gb a.o b.o
issotm@sheik-kitty tmp.661JcLpBHt% xxd out.gb| head
00000000: 2a00 0000 0000 0000 0000 0000 0000 0000  *...............
00000010: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000020: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000030: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000040: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000050: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000060: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000070: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000080: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00000090: 0000 0000 0000 0000 0000 0000 0000 0000  ................
```

Now try again without the `SECTION` line, linking will fail.

I'm not sure whether that's a RGBASM or RGBLINK bug, as I haven't analyzed the .o files that get produced.

**Comment by dbrotz** [comment]
It's because `CreateSymbolTable()` in assign.c loops over the sections when creating symbols. If there are no sections, no symbols are created. It would make more sense to loop over the symbol tables of each object file instead of looping over the sections, because that would match how the data is stored in the object file format.

**Comment by ISSOtm** [comment]
```
issotm@sheik-kitty tmp.eOg9s8PB3i% rgblink -V               
rgblink v0.3.9-29-ga40d599
issotm@sheik-kitty tmp.eOg9s8PB3i% cat a.asm
SECTION "a", ROM0[0]
	db ANSWER
issotm@sheik-kitty tmp.eOg9s8PB3i% cat b.asm
ANSWER = 42
EXPORT ANSWER

issotm@sheik-kitty tmp.eOg9s8PB3i% rgbasm -o a.o a.asm
issotm@sheik-kitty tmp.eOg9s8PB3i% rgbasm -o b.o b.asm
issotm@sheik-kitty tmp.eOg9s8PB3i% rgblink -o out.gb a.o b.o
issotm@sheik-kitty tmp.eOg9s8PB3i%
```
Fixed by #434, apparently.
